File: sekre-backend/internal/domain/types/pagination.go
package types

// PaginationParams holds pagination parameters for repository queries
type PaginationParams struct {
	Limit  int
	Offset int
}

// PaginationResult holds paginated data with total count
type PaginationResult struct {
	TotalCount int
}

// NewPaginationParams creates pagination params from limit and offset
func NewPaginationParams(limit, offset int) PaginationParams {
	return PaginationParams{
		Limit:  limit,
		Offset: offset,
	}
}

File: sekre-mobile/composeApp/src/commonMain/kotlin/org/sekre_mobile/com/domain/model/Pagination.kt
package org.sekre_mobile.com.domain.model

/** Pagination parameters aligned with backend contract (limit/offset). */
data class PaginationParams(
    val limit: Int = 50,
    val offset: Int = 0,
) {
    val page: Int
        get() = (offset / limit) + 1

    val pageSize: Int
        get() = limit
}

/** Paginated result */
data class PaginatedResult<T>(
    val items: List<T>,
    val total: Int,
    val limit: Int,
    val offset: Int,
) {
    val page: Int
        get() = (offset / limit) + 1

    val pageSize: Int
        get() = limit

    val totalPages: Int
        get() = if (limit > 0) (total + limit - 1) / limit else 0

    val hasNextPage: Boolean
        get() = (offset + limit) < total

    val hasPreviousPage: Boolean
        get() = offset > 0
}
